lucee-java/lucee-core/src/lucee/commons/io/res/type/s3/S3ResourceOutputStream.java

```diff
@@ -46,11 +46,11 @@ public void close() throws IOException {
 		ts.close();
 
 		try {
-			s3.put(objectName, ts);
-		} catch (SocketException se) {
-			String msg = StringUtil.emptyIfNull(se.getMessage());
-			if (StringUtil.indexOfIgnoreCase(msg, "Socket closed")==-1) {
-				throw se;
+			if (ts.isMemoryMode()) {
+				s3.put(objectName, ts);
+			} else {
+				s3.put(objectName, ts.getBackingFile());
```

Comment:
If we used `putInputStream` we could lead to another write to a file and since we already have a file we can just pass that . (Using a file as the source will always be easier for the SDK to handle than `InputStream`s)